Implementation for CurvePoint: endpoints + logs + constraints

ALEXD2758 opened the pull request:
Implementation of endpoints in CurvePoint Controller
Implementation CurvePointService
Modification methods CurvePointRepository
Implementation logging with LOG4J for BidListController and CurvePointController
Rectification method "validate" in BidListController and CurvePointController
Rectification of JavaDoc for all controllers (missing from previous version)
Implementation definitive comments for CurvePointController
Implementation Hibernate constraints in BidListModel and CurvePointModel

ALEXD2758 committed:
Implementation CurvePointRepository methods
Implementation CurvePointService class and methods
Implementation CurvePointController methods
Redaction definitive JavaDoc in CurvePointController

ALEXD2758 committed:
Modification of references to the Model layer from controller layer and from repository layer
Implementation CurvePoint/list CurvePoint/update and CurvePoint/add HTML pages
Implementation Hibernate constraints in Model classes: BidListModel & CurvePointModel
Rectification errors present in endpoint BidList

ALEXD2758 committed:
Rectification JavaDoc on controllers (missing from previous version)

akath19 commented on src/main/java/com/alex/poseidon/controllers/BidListController.java:
```diff
@@ -34,6 +38,7 @@
     @RequestMapping("/bidList/list")
     public String home(Model model) {
         model.addAttribute("bidList", bidListService.getAllBids());
+        logger.info("bidList/list : OK");
```
This is missing a GET at the beginning

ALEXD2758 replied:
Done

akath19 commented on src/main/java/com/alex/poseidon/controllers/CurveController.java:
```diff
@@ -1,124 +1,170 @@
-        // TODO: find all Curve Point, add to model
+    public String home(Model model) {
+        model.addAttribute("curvePoint", curvePointService.getAllCurvePoints());
+        logger.info("curvePoint/list : OK");
```
This is missing a GET at the beginning

ALEXD2758 replied:
Done

akath19 commented on src/main/java/com/alex/poseidon/models/BidListModel.java:
```diff
@@ -27,34 +31,46 @@
     @Column(name = "benchmark")
     private String benchmark;
     @Column(name = "bid_list_date")
     private Timestamp bidListDate;
```
What happens in the bid list is before now?

ALEXD2758 replied:
Yes and no:
Yes: I forgot the constraints and logger in Bid list in story_1
No: I didn't pull the right version before, therefore my model classes changed (the name of model classes changed and references to it)

akath19 replied:
Hmmmm....that doesn't solve my question, should this work if the bid_list_date has a date before now?

ALEXD2758 replied:
I had misunderstood your question due to line 34 having a green focus. Sorry.
I have added constraint "@FutureOrPresent" for 3 fields which I judge necessary to have a check on the date (except if revisionDate is required to be filled automatically, not by the client).

akath19 reviewed (changes requested):
Please address comments

ALEXD2758 committed:
Rectification GET home in BidListController & CurveController

ALEXD2758 committed:
Implementation Hibernate constraint "FutureOrPresent" in BidListModel, for bidListDate & revisionDate fields.
Implementation Hibernate constraint "FutureOrPresent" in CurvePointModel, for asOfDate field.

akath19 reviewed (approved):
LGTM
